internal/exec/util/file.go

```diff
@@ -152,7 +153,7 @@ func (u Util) WriteLink(s types.Link) error {
 
 func (u Util) SetPermissions(mode *int, node types.Node) error {
 	if mode != nil {
-		mode := os.FileMode(*mode)
+		mode := os.FileMode(toFileMode(mode))
```

Comment:
`toFileMode` already returns an `os.FileMode`, so we don't need the cast.

Reply:
whoops, yeah... good catch!